src/Components/SmartComponents/SystemsPage/SystemsPage.test.js

```diff
@@ -1,36 +1,64 @@
 import SystemsPage from './SystemsPage';
-import { render } from '@testing-library/react';
+import { render, screen, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { useDispatch, useSelector } from 'react-redux';
 import TestWrapper from '../../../Utilities/TestWrapper';
-import { initialState } from '../../../Store/Reducers/SystemsPageStore';
+import { entitiesInitialState, initialState } from '../../../Store/Reducers/SystemsPageStore';
 import configureStore from 'redux-mock-store';
 
+jest.mock('../../../Helpers/Hooks', () => ({
+    ...jest.requireActual('../../../Helpers/Hooks'),
+    useRbac: () => [[true, true, true, true], false]
+}));
+
 jest.mock("react-redux", () => ({
     ...jest.requireActual("react-redux"),
     useDispatch: jest.fn(),
     useSelector: jest.fn()
 }));
 
+jest.mock('@unleash/proxy-client-react', () => ( {
+    ...jest.requireActual('@unleash/proxy-client-react'),
+    useFlag: () => true, 
+    useFlagsStatus: () => ({ flagsReady: true })
+}));
+
 const customMiddleWare = store => next => action => {
     useSelector.mockImplementation(callback => {
-        return callback({ SystemsPageStore: initialState });
+        return callback({ SystemsPageStore: initialState, entities: entitiesInitialState });
     });
     next(action);
 };
 
 const mockStore = configureStore([customMiddleWare]);
-const store = mockStore(initialState);
+const store = mockStore({ SystemsPageStore: initialState, entities: entitiesInitialState });
 
-describe('SystemsPage', () => {
-    let fetchData = jest.fn();
-    let systems = { data: [], meta: {} };
+beforeEach(() => {  
+    useSelector.mockImplementation(callback => {
+        return callback({ SystemsPageStore: initialState, entities: entitiesInitialState });
+    });
+    useDispatch.mockReturnValue(jest.fn());
+});
 
-    it.skip('should call fetchData function', () => {
+afterEach(() => {
+    useSelector.mockClear();
+    store.clearActions();
+});
```

Comment:
It might be better to put them within the `describe` to ensure the effects of these callbacks don't leak into other tests.